Assets/UI/Scenes/NewGameUI.tscn

```diff
@@ -0,0 +1,735 @@
+[ext_resource path="res://Assets/UI/Images/Buttons/close_h.png" type="Texture" id=12]
+[ext_resource path="res://Assets/UI/Scripts/MapList.gd" type="Script" id=13]
+[ext_resource path="res://Assets/UI/Fonts/SmallFontBoldItalic.tres" type="DynamicFont" id=14]
+[ext_resource path="res://Assets/UI/Images/Logos/uh_no_text.svg" type="Texture" id=15]
+[ext_resource path="res://Assets/UI/Images/Buttons/ok.png" type="Texture" id=16]
+[ext_resource path="res://Assets/UI/Images/Buttons/ok_h.png" type="Texture" id=17]
+
+[sub_resource type="Theme" id=1]
+CheckBox/colors/font_color = Color( 0.136719, 0.0924921, 0.0299072, 1 )
+CheckBox/colors/font_color_disabled = Color( 0.9, 0.9, 0.9, 0.2 )
+CheckBox/colors/font_color_hover = Color( 0.136719, 0.0924921, 0.0299072, 1 )
+CheckBox/colors/font_color_pressed = Color( 0.136719, 0.0924921, 0.0299072, 1 )
+CheckBox/constants/check_vadjust = 0
+CheckBox/constants/hseparation = 4
+CheckBox/fonts/font = null
+CheckBox/icons/checked = null
+CheckBox/icons/radio_checked = null
+CheckBox/icons/radio_unchecked = null
+CheckBox/icons/unchecked = null
+CheckBox/styles/disabled = null
+CheckBox/styles/focus = null
+CheckBox/styles/hover = null
+CheckBox/styles/normal = null
+CheckBox/styles/pressed = null
+Label/colors/Label = Color( 0.136719, 0.0924921, 0.0299072, 1 )
+Label/colors/font_color = Color( 0.133333, 0.0901961, 0.027451, 1 )
+Label/colors/font_color_shadow = Color( 0, 0, 0, 0 )
+Label/constants/line_spacing = 3
+Label/constants/shadow_as_outline = 0
+Label/constants/shadow_offset_x = 1
+Label/constants/shadow_offset_y = 1
+Label/fonts/font = null
+Label/styles/normal = null
+
+[sub_resource type="StyleBoxFlat" id=2]
+content_margin_left = 4.0
+bg_color = Color( 1, 1, 1, 0.227451 )
+border_width_left = 2
+border_width_top = 4
+border_width_right = 4
+border_width_bottom = 3
+border_color = Color( 0.258824, 0.188235, 0.0509804, 1 )
+border_blend = true
+corner_radius_top_right = 5
+corner_radius_bottom_right = 25
+
+[sub_resource type="StyleBoxLine" id=3]
+thickness = 0
+
+[sub_resource type="StyleBoxFlat" id=4]
+content_margin_left = 4.0
+content_margin_right = 10.0
+bg_color = Color( 1, 1, 1, 0.227451 )
+border_width_left = 2
+border_width_top = 4
+border_width_right = 4
+border_width_bottom = 3
+border_color = Color( 0.258824, 0.188235, 0.0509804, 1 )
+border_blend = true
+corner_radius_top_right = 5
+corner_radius_bottom_right = 25
+
+[sub_resource type="GDScript" id=12]
+script/source = "extends LineEdit
+
+func _ready() -> void:
+	text = Global.player_name
+
+func _on_PlayerName_focus_exited() -> void:
+	Global.player_name = text
+"
+
+[sub_resource type="GDScript" id=5]
+script/source = "extends ColorRect
+
+export(int) var color_to_faction
+"
+
+[sub_resource type="GDScript" id=11]
+script/source = "extends HSlider
+
+func _ready() -> void:
+	value = Global.ai_players
+
+func _on_AISlider_value_changed(value: float) -> void:
+	Global.ai_players = value
```

Comment:
This function crashes the game due to `ai_players` being an `int` while `value` is a `float`. Change `value` to be an `int`.